src/libopensc/card-idprime.c

```diff
@@ -206,9 +206,13 @@ static int idprime_process_index(sc_card_t *card, idprime_private_data_t *priv,
 				if (card->type == SC_CARD_TYPE_IDPRIME_V2) {
 					/* The key reference starts from 0x11 and increments by the key id (ASCII) */
 					new_object.key_reference = 0x11 + key_id;
-				} else { /* V3 */
-					/* The key reference starts from 0xF7 and increments by the key id (ASCII) */
-					new_object.key_reference = 0xF7 + key_id;
+				} else {
```

Comment:
I feel that a switch/case should be used here.

Reply:
I've joined v2, v3, and v4 in a switch/case in the else branch, since v1 has a substantially simpler handling